src/lib/dialog/dialog-content-directives.ts

```diff
@@ -56,6 +62,9 @@ export class MatDialogClose implements OnInit, OnChanges {
       // be resolved at constructor time.
       this.dialogRef = getClosestDialog(this._elementRef, this._dialog.openDialogs)!;
     }
+
+    const buttonText = this._elementRef.nativeElement.innerText;
```

Comment:
I had been thinking we could special-case `mat-icon` and ignore its text if it's present, but I realized we can actually just ignore the text if it's a `mat-icon-button`.

Reply:
Done. I also reworded it a bit so it's more clear about what it does.